### PR TITLE
adding set_spark_integration_jar_path() to give ability to users set …

### PR DESCRIPTION
**Pull request by esoroush**
…arbitrary/customized spark_unity.jar

### CODE REVIEW

File: oss_src/unity/python/sframe/data_structures/sframe.py [hunk 1]
```diff
@@ -1641,6 +1631,36 @@ def read_json(cls,
...
+        """
+        global SFRAME_GRAPHLABUTIL_REF
+        if not isinstance(jar_path, str):
+            raise TypeError("The jar path must be str type")
+        if not os.path.isfile(jar_path):
+            raise RuntimeError("Cannot find the file %s" % jar_path)
+        if not jar_path.endswith(".jar"):
+            raise RuntimeError("The jar path must point to a jar file")
+        if SFRAME_GRAPHLABUTIL_REF is not None:
+            raise RuntimeError("The jar path should be set before calling to_rdd(), from_rdd(), or to_spark_dataframe() methods.")
```
Comment (hoytak) [inline oss_src/unity/python/sframe/data_structures/sframe.py]: Is this the correct error message?  I'd be pretty confused, as I'd think this is how to set the jar path.

File: oss_src/unity/python/sframe/data_structures/sframe.py [hunk 2]
```diff
@@ -1641,6 +1631,36 @@ def read_json(cls,
...
+
+    @classmethod
+    def get_spark_integration_jar_path(cls):
+        """
+        The absolute path of the jar file required to enable GraphLab Create's
+        integration with Apache Spark.
+        """
+        if 'RDD_JAR_PATH' not in BINARY_PATHS:
+            raise RuntimeError("Could not find a spark integration jar.  "\
+                "Does your version of GraphLab Create support Spark Integration (is it >= 1.0)?")
```
Comment (hoytak) [inline oss_src/unity/python/sframe/data_structures/sframe.py]: Is it that we cant find it, or is it that things are just not set yet?   Is there any possible way that the resolution of this is just to call set_spark_integration_jar_path?  If so, then this error message should reflect that.

File: oss_src/unity/python/sframe/data_structures/sframe.py [hunk 3]
```diff
@@ -1641,6 +1631,36 @@ def read_json(cls,
...
+        if not isinstance(jar_path, str):
+            raise TypeError("The jar path must be str type")
+        if not os.path.isfile(jar_path):
+            raise RuntimeError("Cannot find the file %s" % jar_path)
+        if not jar_path.endswith(".jar"):
+            raise RuntimeError("The jar path must point to a jar file")
+        if SFRAME_GRAPHLABUTIL_REF is not None:
+            raise RuntimeError("The jar path should be set before calling to_rdd(), from_rdd(), or to_spark_dataframe() methods.")
+
+        BINARY_PATHS['RDD_JAR_PATH'] = jar_path
```
Comment (hoytak) [inline oss_src/unity/python/sframe/data_structures/sframe.py]: Should we add in any checking to make sure it's a correct path?

**Comment by esoroush** [comment]
I talked to Hoyt about the code changes and clarified it. It is good to be merged

**Comment by ylow** [comment]
This is a developer only feature and does not belong at the top-level of the namespace.

**Pull request closed without merging**